src/ring_theory/dedekind_domain.lean

```diff
@@ -945,4 +945,115 @@ begin
     { exact this } },
 end
 
+lemma sup_prime_factor (hI : I ≠ ⊥) (p : ideal T) (hp : p ∈ factors I) (n : ℕ) :
+  p^n ⊔ I = p^(min ((factors I).count p) n) :=
+begin
+  suffices H : factors (p^n) = repeat p n,
+  { rw [sup_eq_prod_inf_factors (p^n) I (pow_ne_zero n (prime_of_factor p hp).ne_zero) hI,
+    ← inf_eq_inter, H, repeat_inf, prod_repeat] },
+    suffices : (factors (p^n)).prod = (repeat p n).prod,
+    { have H₁ : associated (factors (p^n)).prod (repeat p n).prod,
+        rw this,
+      have H'' := prime_factors_unique (prime_of_factor) (by {intros x hx, rw eq_of_mem_repeat hx,
+        exact prime_of_factor p hp}) H₁,
+      rw [associated_eq_eq, multiset.rel_eq] at H'',
+      exact H'' },
+    { rw [prod_factors_eq_self, prod_repeat p n],
+      apply pow_ne_zero n (prime_of_factor p hp).ne_zero },
+end
+
+lemma sup_dvd_of_ge (hI : I ≠ ⊥) (p : ideal T) (hp : p ∈ factors I) (n : ℕ)
+  (hn : n ≥ (factors I).count p) : p^n ⊔ I = p^((factors I).count p) :=
+by rw [sup_prime_factor I hI p hp, min_eq_left hn]
+
+lemma sup_dvd_of_le  (hI : I ≠ ⊥) {p : ideal T} (hp : p ∈ factors I) (n : ℕ)
+  (hn : n < (factors I).count p) : p^n ⊔ I = p^n :=
+by rw [sup_prime_factor I hI p hp, min_eq_right (le_of_lt hn)]
+
 end is_dedekind_domain
+
+section quotient_multiplicity
+noncomputable theory
+open_locale classical
+variables {T : Type*} [integral_domain T] [is_dedekind_domain T] (I : ideal T)
+
+open ideal unique_factorization_monoid
+
+/--The quotient multiplicity of `p` is the least natural number `n` such that the sequence of
+ powers `(p')^m` for `m ≥ n` is constant, where `p'` is the image of `p` in `R/I` -/
+@[simp]
+def quotient_multiplicity (p : ideal T) : ℕ :=
+  Inf {n : ℕ | ∀ m : ℕ, m ≥ n → (map I^.quotient.mk p)^n = (map I^.quotient.mk p)^m}
+
+/--The sequence of powers `(p')^m` is eventually constant, where `p'` is the image of `p` in `R/I`-/
+lemma seq_pow_eventually_constant (hI : I ≠ ⊥) (p : ideal T) (Hp : p ∈ factors I) (n : ℕ)
+ (hn : n ≥ (unique_factorization_monoid.factors I).count  p):
+  (map I^.quotient.mk p)^n = (map I^.quotient.mk p)^((factors I).count p) :=
+begin
+  rw [← map_pow, ← map_pow],
+  have H : map I^.quotient.mk ((p^n)) = map I^.quotient.mk (p^n ⊔ I),
+  { rw [map_sup, map_mk_eq_bot_of_le (le_refl I), sup_bot_eq] },
+  rw sup_dvd_of_ge I hI p Hp n hn at H,
+  exact H,
+end
+
+-- This lemma could maybe be removed as it only shows up once in the following code
+lemma quotient_multiplicity_well_defined (hI : I ≠ ⊥) (p : ideal T) (Hp : p ∈ factors I) :
+  {n : ℕ | ∀ m : ℕ, m ≥ n → (map I^.quotient.mk p)^n = (map I^.quotient.mk p)^m}.nonempty :=
+by {use (factors I).count p, intros m hm, exact (seq_pow_eventually_constant I hI p Hp m hm).symm}
```

Comment:
You could actually use this to define `quotient_multiplicity` computably I think using `nat.find`

Reply:
I did at one point consider using `nat.find` for that definition. What scared me away was having to show that 
`∀ m : ℕ, m ≥ n → (map I^.quotient.mk p)^n = (map I^.quotient.mk p)^m}.nonempty` is a decidable predicate (which is necessary for using `nat.find` ?)

Reply:
Actually I've just figured out how to dos this, I'll make the changes and `git push` them in a bit

Reply:
Done